src/main/java/seedu/address/model/student/UniqueStudentList.java

```diff
@@ -241,52 +493,166 @@ create a class from the start time and check to see if it exceeds the startTime
+                newClass = new Class(currDate.plusDays(1), startTimeFromTr, endTimeFromTr);
+            }
+        } else if (currTime.compareTo(tr.startTimeRange) > 0 && currTime.compareTo(tr.endTimeRange) < 0) {
+            // this means that the currentTime is between the start and end time range
+            if (classToCompare.endTime.compareTo(tr.startTimeRange) <= 0) {
+                if (endTimeFromCurrTime.compareTo(tr.endTimeRange) <= 0) {
+                    newClass = new Class(currDate, startTimeFromCurrTime, endTimeFromCurrTime);
+                }
+            } else if (classToCompare.endTime.compareTo(tr.startTimeRange) > 0
+                    && classToCompare.endTime.compareTo(tr.endTimeRange) < 0
+                    && classToCompare.startTime.compareTo(tr.startTimeRange) <= 0) {
+                if (currTime.compareTo(classToCompare.endTime) <= 0) {
+                    if (endTimeFromClass.compareTo(tr.endTimeRange) <= 0) {
+                        newClass = new Class(currDate, startTimeFromClass, endTimeFromClass);
+                    }
+                } else {
+                    if (endTimeFromCurrTime.compareTo(tr.endTimeRange) <= 0) {
+                        newClass = new Class(currDate, startTimeFromCurrTime, endTimeFromCurrTime);
+                    }
+                }
+            } else if (classToCompare.endTime.compareTo(tr.endTimeRange) < 0
+                    && classToCompare.startTime.compareTo(tr.startTimeRange) > 0) {
+                if (currTime.compareTo(classToCompare.startTime) < 0) {
+                    if (endTimeFromTr.compareTo(classToCompare.startTime) <= 0) {
+                        newClass = new Class(currDate, startTimeFromTr, endTimeFromTr);
+                    }
+                } else if (currTime.compareTo(classToCompare.startTime) >= 0
+                    && currTime.compareTo(classToCompare.endTime) <= 0) {
+                    if (endTimeFromClass.compareTo(tr.endTimeRange) <= 0) {
+                        newClass = new Class(currDate, startTimeFromClass, endTimeFromClass);
+                    }
+                } else if (currTime.compareTo(classToCompare.endTime) > 0) {
+                    if (endTimeFromCurrTime.compareTo(tr.endTimeRange) <= 0) {
+                        newClass = new Class(currDate, startTimeFromCurrTime, endTimeFromCurrTime);
+                    }
+                }
+            } else if (classToCompare.startTime.compareTo(tr.startTimeRange) > 0
+                    && classToCompare.endTime.compareTo(tr.endTimeRange) >= 0) {
+                if (currTime.compareTo(classToCompare.startTime) < 0) {
+                    if (endTimeFromCurrTime.compareTo(classToCompare.startTime) <= 0
+                        && endTimeFromCurrTime.compareTo(tr.endTimeRange) <= 0) {
+                        newClass = new Class(currDate, startTimeFromCurrTime, endTimeFromCurrTime);
+                    }
+                }
+            } else {
```

Comment:
this else block is duplicative of the nullity check in line 644. Same applies for the else block in line 639.

Reply:
Resolved!